src/nvim/CMakeLists.txt

```diff
@@ -120,9 +120,14 @@ if(NOT MSVC)
   set_source_files_properties(
     ${CONV_SOURCES} PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS} -Wno-conversion")
   # gperf generates ANSI-C with incorrect linkage, ignore it.
-  set_source_files_properties(
-    eval.c PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS} -Wno-static-in-inline -Wno-conversion")
-
+  check_c_compiler_flag(-Wno-static-in-inline HAS_WNO_STATIC_IN_INLINE_FLAG)
```

Comment:
AFAIK OpenBSD's gcc is a fork of gcc 4.2 with their custom changes, [gcc-local](http://man.openbsd.org/OpenBSD-current/man1/gcc-local.1).  I believe its kept as default because of the licensing changes from 4.2 to 4.3, and because clang does not support all target architectures ([more on this issue](http://marc.info/?l=openbsd-misc&m=137530560232232)).

There are packages for clang and gcc 4.9 is in ports (but no pkg). Maybe they'll start using clang in base one day, thats what FreeBSD did. But I still come across systems that are stuck with 4.3 packages, and it peeves me a bit that our build requirements would require 4.4 because of a single optional flag.